src/app/application.cpp

```diff
@@ -482,8 +482,18 @@ int Application::exec(const QStringList &params)
     ScanFoldersModel::initInstance(this);
 
 #ifndef DISABLE_WEBUI
+#ifndef DISABLE_GUI
     m_webui = new WebUI;
-#endif
+#else // DISABLE_GUI
+    try {
```

Comment:
You wrote WebUI::WebUI() to throw regardless of DISABLE_GUI, so the try block should also work in both cases.

Reply:
Oops... I overlooked that. I will ifdef the constructor too.

Reply:
Why not unconditional try? Why WebUI should know whether GUI is enabled or not?

Reply:
Because only in nox we terminate. In GUI the user is able to reconfigure.

Reply:
So the logic can be moved into the `catch` clause.